library/src/main/java/com/google/android/exoplayer/text/webvtt/WebvttParser.java

```diff
@@ -164,29 +182,32 @@ public final WebvttSubtitle parse(InputStream inputStream) throws IOException {
 
         try {
           if ("line".equals(name)) {
+            Pair<String, Alignment> lineMetadata = parseLinePositionAttributes(value);
+            value = lineMetadata.first;
+            lineAlignment = lineMetadata.second;
             if (value.endsWith("%")) {
               lineNum = parseIntPercentage(value);
-            } else if (value.matches(NON_NUMERIC_STRING)) {
-              Log.w(TAG, "Invalid line value: " + value);
             } else {
-              lineNum = Integer.parseInt(value);
+              // Following WebVTT spec, line number can be a negative number
+              int sign = 1;
```

Comment:
Yeah, you are right, Integer.parseInt can parse the negative sign. However, doing the check using the regular expression add extra complexity to the regular expression (create a separated regex for line numbers apart from the one used for checking position numbers; the new regex should check string doesn't include numbers nor an initial hyphen character). 

Personally I thought using a new and modified regex for this could make code looks less clean.